zulipterminal/ui_tools/boxes.py

```diff
@@ -1419,6 +1414,44 @@ def indent_quoted_content(soup: Any, padding_char: str) -> int:
             pad_count += 1
         return bq_len
 
+    @staticmethod
+    def parse_codeblock(code_soup: Any) -> List[Any]:
+        markup = []
+
+        code_text = code_soup.text
+        max_line_len = len(max(code_text.split("\n"), key=lambda x: len(x)))
+
+        line_len = 0
+        for code_element in code_soup.contents:
```

Comment:
The logic here looks very complex, and I can't tell if it's doing the correct thing. Can you loop over lines instead?

Reply:
No that is not possible, we have to parse element wise. Sometimes that element can contain only one character only. I think  if we split lines using code_text we would loose out on tag information(css class, element). I can write a docstring explaining the `codeblock` format?